File: Beecrowd/2-AD-HOC/1246-Estacionamento-7/1246.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        string input;
        while ((input = Console.ReadLine()) != null) {
            string[] tokens = input.Split();
            int size = int.Parse(tokens[0]);
            int cases = int.Parse(tokens[1]);

            int[] estacionamento = new int[size];
            Dictionary<int, int> position = new Dictionary<int, int>();
            int ans = 0;

            for (int r = 0; r < cases; r++) {
                string[] cmd = Console.ReadLine().Split();
                char command = cmd[0][0];

                if (command == 'C') {
                    int placa = int.Parse(cmd[1]);
                    int sizeCar = int.Parse(cmd[2]);
                    bool aux;
                    for (int i = 0; i <= size - sizeCar; i++) {
                        if (estacionamento[i] == 0) {
                            aux = true;
                            for (int j = i + 1; j < i + sizeCar; j++) {
                                if (estacionamento[j] > 0) {
                                    aux = false;
                                    break;
                                }
                            }
                            if (aux) {
                                estacionamento[i] = sizeCar;
                                ans++;
                                position[placa] = i;
                                break;
                            }
                        } else {
                            i += estacionamento[i] - 1;
                        }
                    }
                } else {
                    int placa = int.Parse(cmd[1]);
                    if (position.ContainsKey(placa)) {
                        estacionamento[position[placa]] = 0;
                        position.Remove(placa);
                    }
                }
            }
            Console.WriteLine(ans * 10);
        }
    }
}

File: Beecrowd/2-AD-HOC/1246-Estacionamento-7/1246.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNextInt()) {
            int size = sc.nextInt();
            int cases = sc.nextInt();
            int[] estacionamento = new int[size];
            Map<Integer, Integer> position = new HashMap<>();
            int ans = 0;

            for (int r = 0; r < cases; r++) {
                char command = sc.next().charAt(0);
                if (command == 'C') {
                    int placa = sc.nextInt();
                    int sizeCar = sc.nextInt();
                    boolean aux;
                    for (int i = 0; i <= size - sizeCar; i++) {
                        if (estacionamento[i] == 0) {
                            aux = true;
                            for (int j = i + 1; j < i + sizeCar; j++) {
                                if (estacionamento[j] > 0) {
                                    aux = false;
                                    break;
                                }
                            }
                            if (aux) {
                                estacionamento[i] = sizeCar;
                                ans++;
                                position.put(placa, i);
                                break;
                            }
                        } else {
                            i += estacionamento[i] - 1;
                        }
                    }
                } else {
                    int placa = sc.nextInt();
                    if (position.containsKey(placa)) {
                        estacionamento[position.get(placa)] = 0;
                        position.remove(placa);
                    }
                }
            }
            System.out.println(ans * 10);
        }
        sc.close();
    }
}
